src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerUtils.java

```diff
@@ -275,4 +279,56 @@ public static AssemblyResultSet assembleReads(final AssemblyRegion region,
+     * callable region (the ref region over which a caller is using these ReadLikelihoods to call variants)
+     *
+     * @param likelihoodsHaplotype ReadLikelihoods containing reads to be annotated along with haplotypes to which these reads have been aligned
+     * @param callableRegion ref region over which caller is using these ReadLikelihoods to call variants
+     */
+    public static void annotateReadLikelihoodsWithRegions(final ReadLikelihoods<Haplotype> likelihoodsHaplotype,
+                                                          final Locatable callableRegion) {
+        //assign alignment regions to each read
+        final Collection<ReadLikelihoods<Haplotype>.BestAllele> bestHaplotypes = likelihoodsHaplotype.bestAllelesBreakingTies();
+        for (final ReadLikelihoods<Haplotype>.BestAllele bestHaplotype : bestHaplotypes) {
+            final GATKRead read = bestHaplotype.read;
+            final Haplotype haplotype = bestHaplotype.allele;
+            read.setAttribute(ALIGNMENT_REGION_TAG, haplotype.getGenomeLocation().toString());
+        }
+
+        //assign callable region to each read
+        final int sampleCount = likelihoodsHaplotype.numberOfSamples();
+        for (int i = 0; i < sampleCount; i++) {
+            for (final GATKRead read : likelihoodsHaplotype.sampleReads(i)) {
+                read.setAttribute(CALLABLE_REGION_TAG, callableRegion.toString());
+            }
+        }
+    }
+
+    /**
+     * Annotates reads in ReadLiklihoods with genotype supported by each read a particular variant.  If a read already has a
+     * supported genotypes annotation this additional annotation is appended to the previous annotation, it does not replace it.
+     * @param vc The variant for which to annotate the reads
+     * @param likelihoodsAllele ReadLiklihoods containing reads to be annotated along with alleles of the variant vc
+     */
+    public static void annotateReadLikelihoodsWithSupportedGenotypes(final VariantContext vc,
+                                                                     final ReadLikelihoods<Allele> likelihoodsAllele) {
+        //assign supported Genotypes to each read
+        final Set<Allele> alleles = new LinkedHashSet<>(vc.getAlleles());
+        final Map<Allele, List<Allele>> alleleSubset = alleles.stream().collect(Collectors.toMap(a -> a, Arrays::asList));
+        final ReadLikelihoods<Allele> subsettedLikelihoodsAllele = likelihoodsAllele.marginalize(alleleSubset);
+        final Collection<ReadLikelihoods<Allele>.BestAllele> bestAlleles = subsettedLikelihoodsAllele.bestAllelesBreakingTies().stream()
+                .filter(ba -> ba.isInformative()).collect(Collectors.toList());
+        for (ReadLikelihoods<Allele>.BestAllele bestAllele : bestAlleles) {
+            String attribute = "";
+            GATKRead read = bestAllele.read;
+            Allele allele = bestAllele.allele;
+            if (read.hasAttribute(SUPPORTED_GENOTYPES_TAG)) {
```

Comment:
How about
```
final String genotypeString = read.hasAttribute(SUPPORTED_GENOTYPES_TAG) ?
 read.getAttributeAsString(SUPPORTED_GENOTYPES_TAG) + ", " : "";
final String intervalString = vc.getContig() + ":" + vc.getStart() + "=" + vc.getAlleleIndex(allele);
read.setAttribute(SUPPORTED_GENOTYPES_TAG, genotypeString + intervalString);
```

Reply:
I agree, this is a better way to do this